subprojects/execution/src/main/java/org/gradle/internal/execution/history/changes/AbstractFingerprintChanges.java

```diff
@@ -16,14 +16,27 @@
 
 package org.gradle.internal.execution.history.changes;
 
-import org.gradle.internal.change.ChangeContainer;
-import org.gradle.internal.change.ChangeVisitor;
+import com.google.common.collect.ImmutableMap;
 import org.gradle.internal.fingerprint.CurrentFileCollectionFingerprint;
 import org.gradle.internal.fingerprint.FileCollectionFingerprint;
+import org.gradle.internal.fingerprint.FingerprintingStrategy;
+import org.gradle.internal.fingerprint.impl.AbsolutePathFingerprintingStrategy;
+import org.gradle.internal.fingerprint.impl.IgnoredPathFingerprintingStrategy;
+import org.gradle.internal.fingerprint.impl.NameOnlyFingerprintingStrategy;
+import org.gradle.internal.fingerprint.impl.RelativePathFingerprintingStrategy;
 
 import java.util.SortedMap;
 
 public abstract class AbstractFingerprintChanges implements ChangeContainer {
+    private static final ImmutableMap<String, FingerprintCompareStrategy> COMPARE_STRATEGY_MAPPING = ImmutableMap.<String, FingerprintCompareStrategy>builder()
+        .put(AbsolutePathFingerprintingStrategy.IGNORE_MISSING.getIdentifier(), AbsolutePathFingerprintCompareStrategy.INSTANCE)
+        .put(NameOnlyFingerprintingStrategy.INSTANCE.getIdentifier(), NormalizedPathFingerprintCompareStrategy.INSTANCE)
+        .put(RelativePathFingerprintingStrategy.IDENTIFIER, NormalizedPathFingerprintCompareStrategy.INSTANCE)
+        .put(IgnoredPathFingerprintingStrategy.INSTANCE.getIdentifier(), IgnoredPathCompareStrategy.INSTANCE)
+        .put(FingerprintingStrategy.CLASSPATH_IDENTIFIER, ClasspathCompareStrategy.INSTANCE)
+        .put(FingerprintingStrategy.COMPILE_CLASSPATH_IDENTIFIER, ClasspathCompareStrategy.INSTANCE)
```

Comment:
Would be a little nicer I think if all of these would have an `.*IDENTIFIER` field instead of doing `INSTANCE.getIdentifier()` in some cases.

Reply:
I think the best way would be if the fingerprinter registry you provide this map. After all, we want to have the list of fingerprinters pluggable.